### ISSUE TITLE
Fails with Gradle configuration cache enabled

### ISSUE THREAD

**Issue by sergeykad** [jsonschema2dataclass/js2d-gradle#884]
Running on Gradle 8.6 with enabled configuration cache results in the following failure:

```
* What went wrong:
Some problems were found with the configuration of task ':mymodule:generateJsonSchema2DataClassConfigMain' (type 'Js2pGenerationTask').
  - Type 'org.jsonschema2dataclass.js2p.Js2pGenerationTask' property 'js2dConfiguration' doesn't have a configured value.
    
    Reason: This property isn't marked as optional and no value has been configured.
    
    Possible solutions:
      1. Assign a value to 'js2dConfiguration'.
      2. Mark property 'js2dConfiguration' as optional.
    
    For more information, please refer to https://docs.gradle.org/8.6/userguide/validation_problems.html#value_not_set in the Gradle documentation.
  - Type 'org.jsonschema2dataclass.js2p.Js2pGenerationTask' property 'js2dConfigurationPlugins' doesn't have a configured value.
    
    Reason: This property isn't marked as optional and no value has been configured.
    
    Possible solutions:
      1. Assign a value to 'js2dConfigurationPlugins'.
      2. Mark property 'js2dConfigurationPlugins' as optional.
    
    For more information, please refer to https://docs.gradle.org/8.6/userguide/validation_problems.html#value_not_set in the Gradle documentation.

```

**Comment by eirnym** [comment]
Thank you for an issue. 

There was an issue #788 for exactly that. 

Patch wasn't successful though. I'll be glad if you have a proper solution.

**Comment by sergeykad** [comment]
With the latest Java LTS being 21 I am guessing most of the projects are on Java 11+  by now.  
Unfortunately, I do not have experience in Kotlin or writing Gradle plugins, so I probably will not be able to help much. Maybe it's just adding the [@Optional](https://docs.gradle.org/8.6/javadoc/org/gradle/api/tasks/Optional) annotation as mentioned in the documentation?

**Comment by eirnym** [comment]
Kotlin is quite similar to Java even syntax is not the same. Don't worry about writing "perfectly". 

Optional is not the case as information in the configuration is mandatory. 

So it's important to find which variable has not been fulfilled for Gradle. 

There's a disabled test which you can use and add an additional argument to pass a debug argument. In GradleVersions.kt there's listed all gradle versions to be tested, almost all of them can be disabled by commenting out or an environment variable to be passed. 

Could you please help me?

**Comment by eirnym** [comment]
Links to files mentioned

https://github.com/jsonschema2dataclass/js2d-gradle/blob/main/plugin-gradle/plugin/src/test/kotlin/org/jsonschema2dataclass/js2p/JavaTaskFunctionalTest.kt

https://github.com/jsonschema2dataclass/js2d-gradle/blob/main/plugin-gradle/plugin/src/test/kotlin/org/jsonschema2dataclass/js2p/GradleVersions.kt

**Comment by eirnym** [comment]
Hi @sergeykad, I probably fixed the issue. Presumably, configurations must be visible to be cacheable. We'll see how tests will go.

**Comment by sergeykad** [comment]
Hi @eirnym, sorry for nagging, but when do you think you will be able to release version 7? The last release was over a year ago.

**Comment by sergeykad** [comment]
Hi @eirnym , any news regarding when you will release a new version?
This is the only issue that keeps us from using the configuration cache.

**Comment by kristoisberg** [comment]
Any updates regarding this?

**Comment by kristoisberg** [comment]
Since 7.0.0 seems like an ambitious release, would it be possible for a minor version to be released with just this fix? This is also the main thing preventing us from upgrading to the latest version of Gradle.

**Comment by Nouran-11** [comment]
Hi @eirnym , thanks for fixing this issue in the codebase.
Would it be possible to release a new version that includes this fix? This would allow us (and others) to start using the plugin with configuration cache enabled without workarounds. Once released, I think we could also close this issue as resolved.

**Comment by eirnym** [comment]
Yep. I'll release it in few days. But Android probably will be dropped. 

If anyone would like to implement it, I'll accept the patch.
